### PR TITLE
node-github ➜ @octokit/rest

### LINKED ISSUE #677: rename to `@octokit/rest`
As announced in https://github.com/octokit/node-github/issues/620#maintainability, it’s time to take the final step to make this the official Oktokit library for JavaScript.

- [x] #680 refactor code. Now with 100% test coverage, it is finally safe to do so
- [x] replace all mentions of `node-github` / `github` in code with `@octokit/rest`

moved remaining TODOs to the PR at #692 

Please comment here if you have any questions

### PR DESCRIPTION
**Pull request by gr2m**
Follow up for #680, closes #677

Here is an overview of **[How the new `@octokit/rest` works](https://github.com/octokit/node-github/blob/677/rename-to-octokit-rest-clean/HOW_IT_WORKS.md)**.

This PR is ready for review. Let’s get #691 merged first, once done change base for this PR to master.

The tests are currently failing as the examples are run as part of the test suite and some of them expect github.com/octokit/rest.js to exist. They will pass once we rename the repository.

I’ve created a separate module for URL template parsing which is specific to GitHub: https://github.com/gr2m/octokit-rest-url-template/pull/1

I’d like to move that repo to Octokit, too, unless there are any objections? Once `@octokit/rest-url-template` is released, update package.json to install from npm instead of from GitHub

See more todos in #677

### Todos

- [x] review & merge https://github.com/octokit/node-github/pull/691. Change base of this PR to `master`
- [x] code reviews by @bkeepers  & @zeke
- [x] test against @probot’s test suite
- [x] test against @greenkeeperio’s test suite
- [x] clean up commits
- [x]  redirect https://octokit.github.io/node-github/ ➜ https://octokit.github.io/rest.js/ (
- [x] rename `node-github` repository to `rest.js`
- [x] merge this PR
- [x] release `@octokit/rest@14.0.0` (to continue versioning from github@13)
- [ ] _([pending mikedeboer/mikedeboer.github.io#1](https://github.com/mikedeboer/mikedeboer.github.io/pull/1))_ make sure https://mikedeboer.github.io/node-github/ redirects to https://octokit.github.io/rest.js/ – [edit this file](https://github.com/mikedeboer/mikedeboer.github.io/blob/master/node-github/index.html)
- [x] deprecate all versions of `node-github` mentioning the new version
- [x] reset Greenkeeper after the rename

### CODE REVIEW

File: CONTRIBUTING.md
Comment (bkeepers) [inline CONTRIBUTING.md]: We should create a redirect in https://github.com/octokit/octokit.github.com similar to what we did when we first moved node-github to this org.

File: LICENSE
Comment (bkeepers) [inline LICENSE:4]: @mlinksva we adopted `node-github` from the community and are renaming it to `octokit`. Does this look ok?

**Review by bkeepers — DISMISSED** [review]
This looks _really_ good! I love how modular it is, and I'm excited to try out the hooks and plugins API.

I'd love to get another set of eyes on this before merging. @zeke?

File: CONTRIBUTING.md
Comment (gr2m) [reply CONTRIBUTING.md]: I got it in the todos, I moved them from the issue to the PR so they will be harder to miss

File: LICENSE
Comment (mlinksva) [reply LICENSE:4]: 2017-2018? 😁  Looks good to me.

Comment (gr2m) [reply LICENSE:4]: would `2017-present` work, too? I’ve seen that in around in other project’s license files

Comment (mlinksva) [reply LICENSE:4]: `-present` is pretty bogus, it's guaranteed to be false at some point.

Comment (mlinksva) [reply LICENSE:4]: However, feel free to skip the year from the added copyright line if you wish. It isn't required.

**Comment by gr2m** [comment]
I’ve put together a [How it works](https://github.com/octokit/node-github/blob/677/rename-to-octokit-rest-clean/HOW_IT_WORKS.md) document for orientation

File: CONTRIBUTING.md
Comment (TiagoDanin) [reply CONTRIBUTING.md]: Recommend using the URL with `https`

Comment (gr2m) [reply CONTRIBUTING.md]: fixed it, cheers!

**Review by zeke — DISMISSED** [review]
👏 🚢 👏 🚢 👏 🚢

**Comment by gr2m** [comment]
I’ve a pull request ready for @probot: https://github.com/probot/probot/pull/400
I’ve run into errors when testing against @greenkeeperio’s test suite, I will get these sorted out first

**Comment by gr2m** [comment]
~~All right, all set from my side. I’d like to rename the repository next, but we have to setup the redirect https://octokit.github.io/node-github/ ➜ https://octokit.github.io/rest.js/, I don’t know how to do it 🙏~~ PR for the API docs redirect is at https://github.com/octokit/octokit.github.com/pull/9

**Review by bkeepers — APPROVED** [review]
🎉

**Comment by orta** [comment]
🎉 congrats!

**Pull request merged**
